src/docker-images/watchdog/src/watchdog.py

```diff
@@ -356,18 +356,37 @@ def parse_pod_item(pod, k8s_pod_gauge, k8s_container_gauge, pods_info,
 
     # get service name from label
     service_name = labels.get("app") or labels.get("jobId")
+    if labels.get("app") is not None:
+        pod_type = "service"
+    else:
+        pod_type = "job"
 
     annotations = walk_json_field_safe(pod, "metadata", "annotations") or {}
     if host_ip != "unscheduled":
         process_service_endpoints(service_name, host_ip, annotations,
                                   service_endpoints)
 
+    deletion_timestamp = walk_json_field_safe(pod, "metadata",
+                                              "deletionTimestamp")
```

Comment:
Does older k8s (1.9) have the same fields?

Reply:
Yes. Just checked.